src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -334,66 +395,98 @@ internal bool TryUnparse(StringBuilder sb)
             }
         }
 
-        public class ArgumentsCore
+        /// <summary>
+        /// The settings for <see cref="TextLoader"/>
+        /// </summary>
+        public class Options
         {
+            /// <summary>
+            /// Whether the input may include quoted values, which can contain separator characters, colons,
+            /// and distinguish empty values from missing values. When true, consecutive separators denote a
+            /// missing value and an empty value is denoted by \"\". When false, consecutive separators denote an empty value.
+            /// </summary>
             [Argument(ArgumentType.AtMostOnce,
                 HelpText =
                     "Whether the input may include quoted values, which can contain separator characters, colons," +
                     " and distinguish empty values from missing values. When true, consecutive separators denote a" +
                     " missing value and an empty value is denoted by \"\". When false, consecutive separators" +
                     " denote an empty value.",
                 ShortName = "quote")]
-            public bool AllowQuoting = DefaultArguments.AllowQuoting;
+            public bool AllowQuoting = Defaults.AllowQuoting;
 
+            /// <summary>
+            /// Whether the input may include sparse representations.
+            /// </summary>
             [Argument(ArgumentType.AtMostOnce, HelpText = "Whether the input may include sparse representations", ShortName = "sparse")]
-            public bool AllowSparse = DefaultArguments.AllowSparse;
+            public bool AllowSparse = Defaults.AllowSparse;
 
+            /// <summary>
+            /// Number of source columns in the text data. Default is that sparse rows contain their size information.
```

Comment:
>Number [](start = 16, length = 6)

would be  nice to have example
theoretically we can have sparse format like SparseSize separtor pairs of index:value devided by separator
743 1:2 5:-1 8:0.4 742:1 743:2
not sure what would happen if we have 744:7 in that line.

Reply:
See my other comment regarding another PR after discussing the public surface of TextLoader.

---
In reply to: [255807082](https://github.com/dotnet/machinelearning/pull/2417#discussion_r255807082) [](ancestors = 255807082)